### DISCUSSION TITLE
Plugin system for managers and datasources

### DISCUSSION THREAD

**Discussion by reegnz** [renovatebot/renovate#28076] [Suggest an Idea]
### Tell us more.

It's been quite difficult to upstream new managers and datasources, rightfully so, as renovate is in widespread use so the risk of breakage is very high from new contributions.

It would however be great if one could develop their own custom datasources and managers, so that one could integrate renovate with less mainstream systems and tools, and to use those custom datasources and managers together with an official renovate release without having to fork renovate.

**Comment by rarkins** [comment]
New managers and datasources have close to no risk of breaking others, because they're modular. If our expand our "public" API for these so that they can be pluggable, it will restrict us and make us move slower, so we don't plan to do that.
